### ISSUE TITLE
🏛️ [EPIC] : Workspace / Proposal Editor

### ISSUE THREAD

**Issue by mike-mcnulty** [input-output-hk/catalyst-voices#111]
### Summary

Create, craft, and submit proposals for a chance to win funding

### JIRA Epic

https://input-output.atlassian.net/browse/NPG-7772

### Description

The Workspace / Proposal Editor will be accessible only for users with a valid Proposer role registration associated to their connected account. It will guide users as they respond to questions, that build into a proposal that can be submitted for a chance to win funding. via community vote. 

Users will have the ability to create and edit proposals at any time, and they need not be attached to an Event / Category at the outset. The basic set of questions that make up the main part of any proposal will be the Base Template. Once a user is ready to attach a proposal to a specific Event / Category for a chance to win funding, they will make a selection in the editor that adds new contextual questions to the proposal template (eg commitment to open source, budget, etc.). These additional questions will be Conditional Questions. 

Users will edit & save responses to individual questions, and each question will include a version history. Users will be able to understand their current progress through the proposal writing question through a) overall progress tracker based on incomplete vs complete questions, and b) individual status indicators next to each section in the contextual navigation menu (eg not started, in progress, action required, completed)

By default, all drafts will be protected by encryption to create a private workspace for users and their collaborators. Users will have the ability make drafts public, in order to collect public comments and feedback to improve their proposal. Upon submission to a particular Funding Track, all proposals will automatically be made public. 

While inside the Proposal Editor, the contextual navigation on the left rail of the General Interface will reflect the questions in the Base Template and selected Conditional Questions. Each section will serve as a clickable anchor that allows the user to quickly jump between sections for easy editing & iteration. Users will also have the ability to show / hide each question in the Editor to enter Focus Mode that focuses on a particular set of related or interdependent questions. Users will be required to save & quit in order to see the regular navigation menu in the left rail again.

The right rail inside the Proposal Editor will include relevant contextual information that can be toggled on / off by the user depending on their needs. Options may include field description, tips for writing quality responses, examples of best practices, and unresolved comments attached to that section. In addition to curating the right rail content, users will have the ability to close the right rail entirely to expand the working space of the Editor. 

When a user attaches a proposal to a particular event, the user automatically subscribes to notifications for that event that remind them about any relevant tasks or deadlines. Once a submission deadline has passed and a proposal moves into an active Funding Round, then the proposal will be locked from any further editing. 

In a future state, it may be desirable to require users to accept a Funding Agreement / Terms & Conditions at the point of proposal submission. In the MVP, this will be out of scope, and funded projects will accept T&C inside the Milestone Module. 

In addition to question-level system validations of user input to ensure alignment with rules for each response, a proposer level validation will take place at the time of submission to ensure agreement with any Campaign level restrictions, such as number of active proposals by a single user. 

If approved for funding, the proposal will remain locked and progress into post-voting stage of the lifecycle, which will be out of scope for the Voices MVP. If not approved, the proposal will return to a draft state, unlock for editing, and be able to be re-submitted in future Funding Rounds. 

On the primary ProjectCatalyst.io webiste, there should be a contextual call to action that deep links into the Proposal Editor. If the user already has a connected account with the Proposer role, they should be dropped directly into the Editor. If they do not, they should be guided through the related registration wizard, then dropped into the Editor.

### Todo

- [ ] Contextual Navigation
- [ ] Education in Context
- [ ] Base Template
- [ ] Category Learn & Select
- [ ] Conditional Questions
- [ ] Focus Mode / Data Controller
- [ ] Event Notifications
- [ ] Privacy Settings
- [ ] Version History
- [ ] Main Website CTA
- [ ] Editor Menu UI Component #279 
- [ ] Batch Submit UI Component #278  

Addtl tickets to be created:

-  [SAVING PROPOSAL] As a user I want to be able to save | save as ... , so I am certain all my latest changed are saved.
- [CLOSE PROPOSAL] As a user I want to be able to close my proposal without saving anything.
- [NAVIGATION | Proposal segment navigation] As a user I want to be able to jump to a specifc topic by clicking the anchor link in the left menu, so I can navigate and focus with ease. AC: Show related support information. AC: Hidden (eye-off) topics are not clickable / greyed out.
- [NAVIGATION | Set Focus] As a user I want to set focus to a topic, by clicking anywhere on the topic box, so the related information appears in the right sidebar. (Post-mvp, left rail leading to get started?)
- [NAVIGATION | Proposal quality | Writing] As a user I want to start a writing action when I press topic edit, so I can level up the quality of my proposal with my selected comments. AC: Comments filter [Selected | All] AC: Iterations save y/n? AC: Version indication? AC: Can have multiple edits open - appears Save all?
- [PROGRESSION] As a user I want to set a progression / completeness indication for a topic in my proposal, to signal to progression to teammates or myself.
- [FOCUS MODE | Proposal writing/concentration] As a user I want to be able to set my workspace in focus mode, so all unnecessary items are hidden and I can write with maximum concentration. AC: All selected in edit mode + save all.
- [FOCUS ACTIONS | RR Proposal Support] As a user I want to be able to switch between education, comments, and action of a segment, so I can find the information/actions for my needs.
- [FOCUS ACTIONS | RR Proposal Support | Comment select] As a user I want to be able to select comments I find valuabe, so I can process those in edit mode of a segment. AC: Add #selected tag AC: Available for proposers only in all spaces?
- [FOCUS ACTIONS | Proposal quality | Status segment/topic] As a user I want to set the compleness/readiness status of a segment/topic, so I can indicate to myself or my team what the progression of our proposal is.
- [FOCUS ACTIONS | Proposal quality | Proposal engagement] As a user I want to be able to share my current view state, so I can ask for comments on these items by sharing the link through socials.
- OVERALL ACTIONS | Bottom bar ] As a user I want easy access to overall proposal actions, so I can swiftly change access, status or publish my proposal.
- OVERALL ACTIONS | Bottom bar ] As a user I want to toggle on/off all proposal segment with one click, so I can deduct or add topics I want to focus on. AC: Title stays on?
- OVERALL ACTIONS | Bottom bar | Rails toggling ] As a user I want to easily hide side-panels, so I can tailor the interface to my needs. AC: zoom levels?
- [OVERALL ACTIONS | Add/Change Team] As a user I want to easily add/change team setup, so can we work collaboratively toward a quality procposal.
- [Proposal navigation] As a user I want to easily access the workspace menu, so I can switch between my proposals. AC: no edits in progress.
- [FOCUS ACTIONS | Proposal quality | Right rail actions Actions
- [team] Setup segment ownership / Save version as ... / Preflight check... / Share current view for comments / Request feedback round.
- Proposer Dashboard shows latest actions of team mates?

**Comment by mike-mcnulty** [comment]
TODO: invite comments, view/hide collaborator comments

**Comment by mike-mcnulty** [comment]
TOOD: Map to workspace
